api/manager.py

```diff
@@ -13,33 +13,198 @@
+            error = "Price per unit missing."
+        elif not productType:
+            error = "Product type missing."
+
+        # continue if no errors
+        if error is None:
+            try:
+                query = """
+                    REPLACE INTO product (display_name, stock, price_per_unit, product_type)
+                    VALUES (?, ?, ?, ?)
+                    """
+                db.execute(query, (displayName, stock, pricePerUnit, productType))
+                db.commit()
+            except:
+                error = "Error saving values"
+            else:
+                return json.dumps({"success": f"{displayName} has been updated."})
+        else:
+            return json.dumps({"error": error})
         # update record on one specific product
 
     if request.method == "POST":
+        # grab data
+        data = request.get_json()
+        displayName = data["display_name"]
+        stock = data["stock"] if "stock" in data else 0
+        pricePerUnit = data["price_per_unit"]
+        productType = data["product_type"]
+
+        # initiating variable
         db = get_db()
-        # add a new product
+        error = None
+
+        # check data entries
+        if not displayName:
+            error = "Product name is required."
+        elif not pricePerUnit:
+            error = "Price per unti is required."
+        elif not productType:
+            error = "Product type is required."
+
+        # continue if no errors
+        if error is None:
+            try:
+                query = """
+                    INSERT INTO product (display_name, stock, price_per_unit, product_type)
+                    VALUES (?, ?, ?, ?)
+                    """
+                db.execute(query, (displayName, stock, pricePerUnit, productType))
+                db.commit()
+            except:
+                error = "Error saving product."
+            else:
+                return json.dumps({"success": f"{productType} added successfully."})
+        else:
+            return json.dumps({"error": error})
 
 
 @bp.route("/user", methods=["GET", "PUT"])
 def user():
     if request.method == "GET":
+        # grab data
+        data = request.get_json
+        search = data["search"]
+
         db = get_db()
+        error = None
+
+        if not search:
+            return json.dumps({"error": "Missing search term."})
+
         # get more info about a specific user
+        query = """
+            SELECT *
+            FROM user
+            WHERE username = ?  
+            """
+        # The idea is to find a specific user, right? Hence the {search term}
+        user = db.execute(query, (search)).fetchall()
+        print(user)
+
+        if user is None:
+            error = "Not implemented"
+        elif len(user) == 0:
+            error = "Empty table"
+
+        if error:
+            return json.dumps({"error": error})
+        else:
+            return json.dumps({"user": user})
```

Comment:
ok ur running into the same "row is not serializable" error i did, i think beau figured out how to fix it before i got around to it so if u wanna ask him how he solved it